### ISSUE TITLE
[GENERAL SUPPORT]: how to use surrogate_spec.model_configs in GenerationStrategy

### ISSUE THREAD

**Issue by zinc65li** [facebook/Ax#3544]
### Question

Hi,

Thanks for the amazing work, I really like the ax-platform. 

When I use GenerationStrategy, I get warning message like "botorch_model_class is deprecated and will be removed in a future version. Please specify botorch_model_class via `surrogate_spec.model_configs`."  Sorry that I cannot find example in the Ax platform Tuorials and I'm lack of Python skill to figure out how to use surrogate_spec.model_configs to define botorch_model_class, acquisition function, etc.

Would you please give me an example about how to use "surrogate_spec.model_configs" in GenerationStrategy and what options are available in the model_configs? Thank you.

### Please provide any relevant code snippet if applicable.

```shell

```

### Code of Conduct

- [x] I agree to follow this Ax's Code of Conduct

**Comment by bernardbeckerman** [comment]
Hi @zinc65li, thanks for raising this here! To be sure I'm giving you relevant feedback, would you please provide a minimal reproducible example that generates the above?

**Comment by zinc65li** [comment]
Hi @bernardbeckerman , thanks for the help. Here is an example code

import torch
from ax.service.ax_client import AxClient, ObjectiveProperties
from ax.modelbridge.generation_strategy import GenerationStep, GenerationStrategy
from ax.modelbridge.registry import ModelRegistryBase, Models
from ax.models.torch.botorch_modular.surrogate import Surrogate
from botorch.models import SaasFullyBayesianSingleTaskGP
from botorch.acquisition.multi_objective.logei import qLogNoisyExpectedHypervolumeImprovement


gs = GenerationStrategy(
    steps=[
        GenerationStep(
            model=Models.SOBOL,
            num_trials=5,
            min_trials_observed=-1,
            max_parallelism=5,
            model_kwargs={"seed": 42},
        ),
        GenerationStep(
            model=Models.BOTORCH_MODULAR,
            num_trials=-1,
            max_parallelism=12,
            should_deduplicate=True,
            model_kwargs={
                "surrogate": Surrogate(
                    botorch_model_class=SaasFullyBayesianSingleTaskGP,
                    mll_options={
                        "num_samples": 64,
                        "warmup_steps": 128,
                        },
                    ),
                "botorch_acqf_class": qLogNoisyExpectedHypervolumeImprovement,
                }
        ),
     
    ]
)

Omit the remaining code.

I get a warning message 
DeprecationWarning: botorch_model_class is deprecated and will be removed in a future version. Please specify botorch_model_class via `surrogate_spec.model_configs`.
  "surrogate": Surrogate(

The code works without issue. I'm just willing to learn how I can modify the code above with `surrogate_spec.model_configs` as suggested by the warning message.
Sorry that I'm lack of Python skill and cannot find an example about how to use the surrogate_spec.model_configs. Would you please help me to get an example of how to use the surrogate_spec.model_configs.? Thank you.

**Comment by bernardbeckerman** [comment]
Please transition from using

```
GenerationStep(
    ...,
    model_kwargs={
        "surrogate": Surrogate(
            botorch_model_class=SaasFullyBayesianSingleTaskGP,
            mll_options={...},
        ),
    },
)
```

to

```
GenerationStep(
    ...,
    model_kwargs={
        "surrogate_spec": SurrogateSpec(
            model_configs=[
                ModelConfig(
                    botorch_model_class=SaasFullyBayesianSingleTaskGP,
                    mll_options={...},
                )
            ]
        )
    }
)
```

I hope that's helpful! Closing this out but please feel free to comment/reopen for more assistance here (or open a new issue for assistance on another topic).

**Comment by zinc65li** [comment]
Great. Thank you so much for your support!
